File: leet_SLWD_209_minsize_subarr_sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums)
    {
        bool    isFound = false;
        int     n = nums.size();
        int     res = 1e6;
        int     sum = 0;
        int     fast = -1;
        int     slow = 0;
        while (++fast < n)
        {
            sum += nums[ fast ];
            while (sum >= target)
            {
                isFound = true;
                res = min(res, fast - slow + 1);
                sum -= nums[slow];
                ++slow;
            }
        }
        if (isFound)
            return res;
        return 0;
    }
};

File: leet_SLWD_209_minsize_subarr_sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        if target in nums:
            return 1
        yes = False
        res = 1e9
        fast = 0
        slow = 0
        add = 0
        while fast < len(nums):
            add += nums[fast]
            while add >= target:
                yes = True
                res = min(res, fast - slow + 1)
                add -= nums[slow]
                slow += 1
            fast += 1
        if yes:
            return res
        return 0
